Add in-app light-box for viewing images.

Sam1301 opened the pull request:
Fixes: #264

Added `LightBoxScreen`to 
* zoom-in/out 
* share
* download images

Sam1301 commented:
Made some additional layout changes (https://drive.google.com/open?id=0B4jK5QX65b0Zek8wWkR5UFllNEE).

Sam1301 commented on src/light-box/LightBox.js:
```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
```
Added an optional right button to ModalNavBar.

Sam1301 commented on src/message/html/HtmlTagImg.js:
```diff
@@ -10,10 +13,32 @@ const styles = StyleSheet.create({
+    this.onPress = this.onPress.bind(this);
+
+    const { src, auth } = this.props;
+    this.fullUrl = getResource(src, auth);
```
yeah, this was a typo, I had meant to add it to props. But using it directly in render certainly makes sense.

Sam1301 commented on src/message/html/HtmlTagImg.js:
```diff
@@ -10,10 +13,32 @@ const styles = StyleSheet.create({
+    this.fullUrl = getResource(src, auth);
+  }
+
+  onPress() {
```
understood 👍

timabbott commented:
Mostly a sidenote, but @borisyankov does it make sense to switch the coveralls configuration to just do the "checks" / status update part, and not also the PR comments?  They're kinda spammy.

borisyankov commented:
It absolutely does, @timabbott I think it was a form of learned helplessness to endure these messages when they don't make sense.

borisyankov commented:
I think we want to get the React Navigation PR in, before merging this.
Let's try to have it soon to master, then resume work on this PR.

Sam1301 commented:
Updated! Now share, download and zoom works on both iOS and Android. Also reworked previous UI.

Android
![screenshot_20170531-174123](https://cloud.githubusercontent.com/assets/14239866/26633047/c166313e-462f-11e7-8406-fca0bd60e583.png)
![screenshot_20170531-174200](https://cloud.githubusercontent.com/assets/14239866/26633053/c37059aa-462f-11e7-9c00-4aa71381db2a.png)
![screenshot_20170531-174223](https://cloud.githubusercontent.com/assets/14239866/26633111/ea0da9c8-462f-11e7-82d4-f104d4d4d027.png)

iOS
<img width="373" alt="screen shot 2017-05-31 at 6 30 56 pm" src="https://cloud.githubusercontent.com/assets/14239866/26633082/d9ee8512-462f-11e7-9fd2-bbc3ffc678ca.png">
<img width="371" alt="screen shot 2017-05-31 at 6 31 16 pm" src="https://cloud.githubusercontent.com/assets/14239866/26633090/dd949f58-462f-11e7-8591-a5c281735bdd.png">
<img width="373" alt="screen shot 2017-05-31 at 6 31 29 pm" src="https://cloud.githubusercontent.com/assets/14239866/26633096/e0a00368-462f-11e7-9ca5-9d05d396d953.png">

borisyankov commented:
Cool. That covers a lot of things... which is also a concern.
We will want to split it in several PRs?

Sam1301 commented:
completely agree 👍

Sam1301 commented:
before splitting this, I am working on two issues the current version has
1) there is some internal error in react-native-fetch-blob which is causing the tests to fail. https://github.com/itinance/react-native-fs could be a better alternative (studying it).
2) adding some tests for light-box dir

borisyankov commented:
Great. Keep it up then ;)

borisyankov commented:
Is this PR still relevant? Anything left unmerged?

Sam1301 commented:
nope, everything is merged 👍
